chain-signatures/node/src/contract_updater.rs

```diff
@@ -24,28 +24,29 @@ impl ContractUpdater {
         contract_state: Arc<RwLock<Option<ProtocolState>>>,
         config: Arc<RwLock<Config>>,
     ) -> anyhow::Result<()> {
-        let mut last_state_update = Instant::now();
-        let mut last_config_update = Instant::now();
         loop {
-            if last_state_update.elapsed() > Duration::from_millis(1000) {
-                let mut contract_state = contract_state.write().await;
-                *contract_state =
-                    rpc_client::fetch_mpc_contract_state(&self.rpc_client, &self.mpc_contract_id)
-                        .await
-                        .ok();
-                last_state_update = Instant::now();
-            }
-            if last_config_update.elapsed() > Duration::from_millis(300) {
+            {
+                match rpc_client::fetch_mpc_contract_state(&self.rpc_client, &self.mpc_contract_id)
+                    .await
+                {
+                    Err(error) => {
+                        tracing::error!("could not fetch contract's state: {error:?}");
+                    }
+                    Ok(state) => {
+                        let mut contract_state_guard = contract_state.write().await;
+                        *contract_state_guard = Some(state);
+                    }
+                }
+
                 let mut config = config.write().await;
                 if let Err(error) = config
                     .fetch_inplace(&self.rpc_client, &self.mpc_contract_id)
                     .await
                 {
                     tracing::error!("could not fetch contract's config: {error:?}");
                 }
-                last_config_update = Instant::now();
             }
-            tokio::time::sleep(Duration::from_millis(100)).await;
+            tokio::time::sleep(Duration::from_millis(1000)).await;
```

Comment:
I guess it doesn't matter that we're doing config update now every 1s, but we should eventually optimize the polling to get both config and state in one RPC call, but that can be done in another PR